CMS-Enterprise/mint-app
# [MINT-3199] Custom Category Form Modal

# MINT-3199

## Description


* Added `CategoryForm` component and test and copy
* Added `CreateMTOCategory` and `GetMTOCategories` query and its types
* Updated css and components to the generic `Modal` component
* Added temporary button section for the Custom Category Modal in `ModelToOperations/Home/index.tsx`



## How to test this change

1.  navigate to MTO Home
2. Click on `Add Custom Category` button
3. Verify that the design matches figma
4. Ensure that creation of primary category works
5. After creating a primary category, create another custom category and in the dropdown, check to see if previous newly created category is in the dropdown

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
